security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -325,9 +328,13 @@ func (s *sdsservice) StreamSecrets(stream sds.SecretDiscoveryService_StreamSecre
 			// In ingress gateway agent mode, if the first SDS request is received but kubernetes secret is not ready,
 			// wait for secret before sending SDS response. If a kubernetes secret was deleted by operator, wait
 			// for a new kubernetes secret before sending SDS response.
-			if s.st.ShouldWaitForIngressGatewaySecret(conID, resourceName, token) {
-				sdsServiceLog.Warnf("%s waiting for ingress gateway secret for proxy %q\n", conIDresourceNamePrefix, discReq.Node.Id)
-				continue
+			if !s.fileMountedCerts {
```

Comment:
1. To improve code readability, can you add a comment to explain the difference between the file mounted cert and the ingress gateway secret? 
2. To improve code readability, can you add a comment to explain whether ingress gateway secret is also mounted liked the file mounted cert or ingress gateway secret is provisioned through a different mechanism?

Reply:
@lei-tang do you mean this comment? I have updated it. Could you please let me know if I am missing any thing?

Reply:
The missing items are:
1) Can you add a comment to explain the difference between the file mounted cert and the ingress gateway secret?
2) Can you add a comment to explain whether ingress gateway secret is also mounted liked the file mounted cert or ingress gateway secret is provisioned through a different mechanism?

Reply:
It is there in the comment already, I tweaked the comment a little bit to make it more clear. Let me know if you think any thing else needed to be clarified.